react-app/src/utils/formOrigin.test.ts

```diff
@@ -0,0 +1,103 @@
+// formOrigin.test.ts
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
+import { getFormOrigin, ORIGIN, DASHBOARD_ORIGIN, DETAILS_ORIGIN, SPA_SUBMISSION_ORIGIN, WAIVER_SUBMISSION_ORIGIN } from './formOrigin'; 
+import { Authority } from 'shared-types/authority';
+import { getDashboardTabForAuthority } from './crumbs'; 
+
+// Mock the getDashboardTabForAuthority function
+vi.mock('./crumbs', () => ({
+  getDashboardTabForAuthority: vi.fn(),
+}));
```

Comment:
Do we need mocking?

Reply:
without mocking it the   "should return the correct pathname and search for DASHBOARD_ORIGIN" returns false